app/components/user/TwoFactor/configure.js

```diff
@@ -0,0 +1,325 @@
+  'Google Authenticator': 'https://support.google.com/accounts/answer/1066447'
+};
+
+const AUTHENTICATOR_LIST = (
+  Object.keys(AUTHENTICATORS).map((authenticator_name) => (
+    <a
+      className="blue hover-navy text-decoration-none hover-underline"
+      key={authenticator_name}
+      href={AUTHENTICATORS[authenticator_name]}
+      target="_blank"
+      rel="noopener noreferrer"
+    >
+      {authenticator_name}
+    </a>
+  )).reduce((acc, link, index, items) => {
+    if (index > 0) {
+      if (index < items.length - 1) {
+        acc.push(', ');
+      }
+
+      if (index === items.length - 1) {
+        acc.push(' and ');
+      }
+    }
+
+    acc.push(link);
+
+    return acc;
+  }, [])
+);
+
+class TwoFactorConfigure extends React.PureComponent<Props, State> {
+  state = {
+    generatingTOTP: true,
+    generatedTOTP: false,
+    activatingTOTP: false,
+    activatedTOTP: false,
+    totpId: null,
+    provisioningUri: null,
+    totpToken: ''
+  };
+
+  render() {
+    return (
+      <DocumentTitle title={`Configure Two-Factor Authentication`}>
+        <div className="container">
+          <PageHeader>
+            <PageHeader.Icon>
+              <Icon
+                icon="placeholder"
+                style={{ width: 34, height: 34, marginTop: 3, marginLeft: 3 }}
+              />
+            </PageHeader.Icon>
+            <PageHeader.Title>
+              Configure Two-Factor Authentication
+            </PageHeader.Title>
+            <PageHeader.Description>
+              Manage your two-factor authentication settings.
+            </PageHeader.Description>
+            <PageHeader.Menu>
+              <Button
+                theme="default"
+                outline={true}
+                link="/user/two-factor"
+              >
+                {this.state.activatedTOTP ? 'Done' : 'Cancel'}
+              </Button>
+            </PageHeader.Menu>
+          </PageHeader>
+
+          {this.renderCurrentStatus()}
+        </div>
+      </DocumentTitle>
+    );
+  }
+
+  renderCurrentStatus() {
+    if (this.state.generatingTOTP) {
+      return (
+        <Panel>
+          <Panel.Section>
+            Be very, very quiet - weʼre generating secrets…
+          </Panel.Section>
+        </Panel>
+      );
+    } else if (this.state.generatedTOTP && !this.state.activatedTOTP) {
+      return (
+        <Panel>
+          <Panel.Section>
+            {this.props.viewer.totp && (
+              <Panel className="orange border-orange">
+                <Panel.Section>
+                  <strong>Youʼre about to reconfigure two-factor authentication.</strong> This will replace your existing two-factor authentication applications and recovery codes.
+                </Panel.Section>
+              </Panel>
+            )}
+
+            <p>To {this.props.viewer.totp ? 'reconfigure' : 'activate'} two-factor authentication, scan this QR Code with your authenticator application.</p>
+
+            <QRCode
```

Comment:
How much does the `QRCode` gear bloat the vendor package size? Do we need to delay load it in (like we do code mirror?) or is it kinda meh?

Reply:
I don’t think it’s too significant, but I’ll check!

Reply:
This adds 9.7kb to our gzipped (actual transferred) size.
(comparing https://buildkite.com/buildkite/frontend/builds/2821 with https://buildkite.com/buildkite/frontend/builds/2818 using their bundle analyser output)

This is 1/5 the size of codemirror, and also codemirror seems to be in the main bundle anyway? Hmm.